dot/parachain/candidate-validation/candidate_validation.go

```diff
@@ -74,7 +85,42 @@ func (cv *CandidateValidation) processMessages(wg *sync.WaitGroup) {
 			case ValidateFromChainState:
 				// TODO: implement functionality to handle ValidateFromChainState, see issue #3919
 			case ValidateFromExhaustive:
-				// TODO: implement functionality to handle ValidateFromExhaustive, see issue #3547
+				result, err := validateFromExhaustive(cv.ValidationHost, msg.PersistedValidationData,
+					msg.ValidationCode, msg.CandidateReceipt, msg.PoV)
+				if err != nil {
+					logger.Errorf("failed to validate from exhaustive: %w", err)
+					msg.Ch <- parachaintypes.OverseerFuncRes[ValidationResult]{
+						Data: ValidationResult{},
+						Err:  err,
+					}
+				} else {
+					msg.Ch <- parachaintypes.OverseerFuncRes[ValidationResult]{
+						Data: *result,
+					}
+					//if err != nil {
+					//	logger.Errorf("failed to validate from exhaustive: %w", err)
+					//	msg.Ch <- parachaintypes.OverseerFuncRes[ValidationResult]{
+					//		Data: ValidationResult{IsValid: false, Err: err},
+					//		Err:  nil,
+					//	}
+					//} else {
+					//	msg.Ch <- parachaintypes.OverseerFuncRes[ValidationResult]{
+					//		Data: ValidationResult{
+					//			IsValid: true,
+					//			CandidateCommitments: parachaintypes.CandidateCommitments{
+					//				UpwardMessages:            result.UpwardMessages,
+					//				HorizontalMessages:        result.HorizontalMessages,
+					//				NewValidationCode:         result.NewValidationCode,
+					//				HeadData:                  result.HeadData,
+					//				ProcessedDownwardMessages: result.ProcessedDownwardMessages,
+					//				HrmpWatermark:             result.HrmpWatermark,
+					//			},
+					//			PersistedValidationData: msg.PersistedValidationData,
+					//			Err:                     nil,
+					//		},
+					//	}
```

Comment:
is this supposed to be commented out?

Reply:
Yes, it's been removed (was leftover test code).